evalml/pipelines/pipeline_base.py

```diff
@@ -66,9 +67,14 @@ def __init__(
         self.random_seed = random_seed
 
         if isinstance(component_graph, list):  # Backwards compatibility
-            self.component_graph = ComponentGraph().from_list(
-                component_graph, random_seed=self.random_seed
-            )
+            with warnings.catch_warnings():
```

Comment:
I added this here to filter for the DeprecationWarning because users who are using AutoMLSearch or PipelineBase might be confused as to why they're receiving this. I think the warning should only surface if the user directly initializes a ComponentGraph using a list directly, Pipelines should still handle lists of components :)

Reply:
Great call